### ISSUE TITLE
The model tend to repeat its generation without proper ending

### ISSUE THREAD

**Issue by ethanyanjiali** [mlfoundations/open_flamingo#147]
Hi team

Thanks for sharing the model and source code. This is a really exciting moment for open source community.

I tried your example quickly, and realized that it tends to repeat the generated text. For example I got this:
```
<image>  An image of two cats. <|endofchunk|> <image>  An image of a bathroom sink. <|endofchunk|> <image>  An image of a Christmas party spread. An image of a Christmas tree. An image of a Christmas tree. An image of a Christmas tree. An image of a Christmas tree. An image of a Christmas tree. An image of a Christmas tree. An image of
```
I tried other examples or prompts and ran into a similar behavior. Do you have any insights on the issue here? I could set `no_repeat_ngram_size` but it will start to make up non-factual things from the image when giving the caption.

**Comment by anas-awadalla** [comment]
Hi @ethanyanjiali,

Yes I have run into this issue. I don’t have a clean fix but here is what we currently [do](https://github.com/mlfoundations/open_flamingo/issues/126). It may be the with more training this issue goes away or that there is an issue in our training setup. I will keep you posted either way.

**Comment by ethanyanjiali** [comment]
I see. Thanks for the quick response. Yes I tried that approach too, looks like that's also how you do the COCO captioning eval. One issue with this approach is that it appears to be less descriptive somehow. For example:
![image](https://user-images.githubusercontent.com/8679679/229269690-579c451b-4dca-43d4-a0dc-86b1efb1aa92.png)
Use "Output:"
```
Output:A japanese dog with a smile on his face.
```
Use "An image of"
```
An image of a smiling Shiba Inu dog.
```


But when I use "An image of", I also noticed that the model tend to hallucinate more often than BLIP-2
